addons/website_blog/controllers/main.py

```diff
@@ -60,6 +60,8 @@ def _prepare_blog_values(self, blogs, blog=False, date_begin=False, date_end=Fal
         """ Prepare all values to display the blogs index page or one specific blog"""
         BlogPost = request.env['blog.post']
         BlogTag = request.env['blog.tag']
+        date_begin = date_begin if isinstance(date_begin, fields.datetime) else False
+        date_end = date_end if isinstance(date_end, fields.datetime) else False
```

Comment:
date_begin = isinstance(date_begin, fields.datetime) and date_begin
date_end = isinstance(date_end, fields.datetime) and date_end